exasol/analytics/schema/column_types.py

```diff
@@ -0,0 +1,119 @@
+import re
+from dataclasses import dataclass
+from enum import (
+    Enum,
+    auto,
+)
+from typing import (
+    Any,
+    Callable,
+    Optional,
+)
+
+
+class CharSet(Enum):
+    UTF8 = auto()
+    ASCII = auto()
+
+    @classmethod
+    def from_string(cls, name: str) -> "CharSet":
+        for c in cls:
+            if c.name == name:
+                return c
+        raise ValueError(f"Couldn't find CharSet with name '{name}'")
+
+
+class PyexasolOption(Enum):
+    CHARACTER_SET = "characterSet"
+    PRECISION = "precision"
+    SCALE = "scale"
+    SIZE = "size"
+    SRID = "srid"
+    UNIT = "unit"
+    WITH_LOCAL_TIME_ZONE = "withLocalTimeZone"
+
+
+@dataclass(frozen=True)
+class PyexasolMapping:
+    int_keys: list[PyexasolOption]
+    modifier_key: Optional[PyexasolOption] = None
+    modifier_getter: Callable[[dict[str, Any]], str] = lambda x: ""
+
+    def modifier(self, args: dict[str, Any]) -> str:
+        return (
+            str(args.get(self.modifier_key.value, ""))
+            if self.modifier_key
+            else self.modifier_getter(args)
+        )
+
+
+class ColumnTypeSource(Enum):
+    SQL = auto()
+    PYEXASOL = auto()
+
+
+@dataclass(frozen=True)
+class SqlType:
+    """
+    Represents an SQL column type, examples:
+
+    * SqlType("VARCHAR", [100], "ASCII")
+    * SqlType("DECIMAL", [10,2], "")
+    * SqlType("HASHTYPE", [8], "BIT")
+    * SqlType("TIMESTAMP", [], "WITH LOCAL TIME ZONE")
+    """
+
+    name: str
+    int_args: list[int]
+    modifier: str
+    source: ColumnTypeSource = ColumnTypeSource.SQL
+
+    def int_dict(self, keys: list[str]) -> dict[str, Any]:
+        return dict(zip(keys, self.int_args))
+
+    @property
+    def char_type_args(self) -> dict[str, Any]:
+        args: dict[str, Any] = dict(zip(["size"], self.int_args))
```

Comment:
This is equivalent of
`args = {"size": self.int_args[0]} if self.int_args else {}`
I guess the case `not self.int_args` is invalid, so we should either raise an exception or just use `{"size": self.int_args[0]}`

Reply:
See next push.